File: isegoria/src/main/java/com/eulersbridge/isegoria/vote/VoteDoneFragment.java
package com.eulersbridge.isegoria.vote;

import android.arch.lifecycle.ViewModelProviders;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.eulersbridge.isegoria.R;
import com.eulersbridge.isegoria.util.ui.TitledFragment;

public class VoteDoneFragment extends Fragment implements TitledFragment {

    private Button addToCalendarButton;
    private VoteViewModel viewModel;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.vote_fragment_done, container, false);

        addToCalendarButton = rootView.findViewById(R.id.vote_done_button_add_to_calendar);
        addToCalendarButton.setEnabled(false);
        addToCalendarButton.setOnClickListener(view -> addToCalendar());

        if (getParentFragment() != null) {
            viewModel = ViewModelProviders.of(getParentFragment()).get(VoteViewModel.class);

            viewModel.getLatestVoteReminder().observe(this, success -> {
                if (success != null && success)
                    addToCalendarButton.setEnabled(true);
            });
        }

        return rootView;
    }


    private void addToCalendar() {
        if (viewModel != null)
            return;

        addToCalendarButton.setEnabled(false);

        Intent addToCalendarIntent = viewModel.getAddVoteReminderToCalendarIntent();

        if (addToCalendarIntent != null
                && getActivity() != null
                && addToCalendarIntent.resolveActivity(getActivity().getPackageManager()) != null)
            getActivity().startActivity(addToCalendarIntent);

        addToCalendarButton.setEnabled(true);
    }

    @Override
    public String getTitle(Context context) {
        return context.getString(R.string.vote_tab_3);
    }
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/vote/VoteDoneFragment.kt
package com.eulersbridge.isegoria.vote

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.eulersbridge.isegoria.R
import com.eulersbridge.isegoria.util.extension.observeBoolean
import com.eulersbridge.isegoria.util.ui.TitledFragment
import kotlinx.android.synthetic.main.vote_fragment_done.*

class VoteDoneFragment : Fragment(), TitledFragment {

    private lateinit var viewModel: VoteViewModel

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View?
            = inflater.inflate(R.layout.vote_fragment_done, container, false)

    fun setViewModel(viewModel: VoteViewModel) {
        this.viewModel = viewModel
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        addToCalendarButton.apply {
            isEnabled = false
            setOnClickListener { addToCalendar() }
        }

        observeBoolean(viewModel.getLatestVoteReminder()) { success ->
            if (!success)
                addToCalendarButton.isEnabled = true
        }
    }

    private fun addToCalendar() {
        addToCalendarButton.isEnabled = false

        activity?.let { activity ->
            viewModel.getAddReminderToCalendarIntent()
                ?.takeIf { it.resolveActivity(activity.packageManager) != null }
                ?.apply {
                    activity.startActivity(this)
                }
        }

        addToCalendarButton.isEnabled = true
    }

    override fun getTitle(context: Context?) = context?.getString(R.string.vote_tab_3)
}